ecs-cli/modules/cli/servicediscovery/servicediscovery_app.go

```diff
@@ -93,28 +129,202 @@ func create(c *cli.Context, networkMode, serviceName, clusterName string, cfnCli
+	}).Infof(overrideMsg, val, override)
+}
+
+// Flags override the values specified in ECS Params
+// Merges fields for Namespace and SDS
+// This function just merges fields; it doesn't validate them
+func mergeSDFlagsAndInput(c *cli.Context, ecsParamsSD *utils.ServiceDiscovery) (*utils.ServiceDiscovery, error) {
+	if ecsParamsSD == nil {
+		ecsParamsSD = &utils.ServiceDiscovery{}
+	}
+
+	// Private DNS Namespace fields
+	privNamespace := ecsParamsSD.PrivateDNSNamespace
+	privNamespace.ID = resolveStringFieldOverride(c, flags.PrivateDNSNamespaceIDFlag, privNamespace.ID, "private_dns_namespace.id")
+	privNamespace.Name = resolveStringFieldOverride(c, flags.PrivateDNSNamespaceNameFlag, privNamespace.Name, "private_dns_namespace.name")
+	privNamespace.VPC = resolveStringFieldOverride(c, flags.VpcIdFlag, privNamespace.VPC, "private_dns_namespace.vpc")
+	ecsParamsSD.PrivateDNSNamespace = privNamespace
+
+	// Public DNS Namespace fields
+	pubNamespace := ecsParamsSD.PublicDNSNamespace
+	pubNamespace.ID = resolveStringFieldOverride(c, flags.PublicDNSNamespaceIDFlag, pubNamespace.ID, "public_dns_namespace.id")
+	pubNamespace.Name = resolveStringFieldOverride(c, flags.PublicDNSNamespaceNameFlag, pubNamespace.Name, "public_dns_namespace.name")
+	ecsParamsSD.PublicDNSNamespace = pubNamespace
+
+	// SDS fields
+	sds := ecsParamsSD.ServiceDiscoveryService
+	sds.DNSConfig.Type = resolveStringFieldOverride(c, flags.DNSTypeFlag, sds.DNSConfig.Type, "dns_config.type")
+	ttl, err := resolveIntPointerFieldOverride(c, flags.DNSTTLFlag, sds.DNSConfig.TTL, "dns_config.ttl")
+	if err != nil {
 		return nil, err
 	}
+	sds.DNSConfig.TTL = ttl
+	threshold, err := resolveIntPointerFieldOverride(c, flags.HealthcheckCustomConfigFailureThresholdFlag, sds.HealthCheckCustomConfig.FailureThreshold, "failure_threshold")
+	if err != nil {
+		return nil, err
+	}
+	sds.HealthCheckCustomConfig.FailureThreshold = threshold
+	ecsParamsSD.ServiceDiscoveryService = sds
 
-	logrus.Info("Waiting for the Service Discovery Service to be created...")
-	// Wait for stack creation
-	cfnClient.WaitUntilCreateComplete(sdsStackName)
+	// top level fields
+	// these container fields aren't used when creating Route53 resources in this package, but we aggregate them here so that we can do error checking- SRV records require these.
+	ecsParamsSD.ContainerName = resolveStringFieldOverride(c, flags.ServiceDiscoveryContainerNameFlag, ecsParamsSD.ContainerName, "container_name")
+	port, err := resolveIntPointerFieldOverride(c, flags.ServiceDiscoveryContainerPortFlag, ecsParamsSD.ContainerPort, "container_port")
+	if err != nil {
+		return nil, err
+	}
+	ecsParamsSD.ContainerPort = port
 
-	// Return the ID of the SDS we just created
-	return getOutputIDFromStack(cfnClient, sdsStackName, CFNTemplateOutputSDSARN)
+	return ecsParamsSD, nil
+}
+
+// Validates that SD input does not container conflicting values
+func validateSDInputFields(input *utils.ServiceDiscovery, networkMode string) error {
```

Comment:
Can we add these helper functions to a separate file in this pkg so that this file is focused on creating and executing the creation of an SD service? (or later, updating one?)